### PR TITLE
Fix return type of `config()`, add documentation

### PR DESCRIPTION
**Pull request by borekb**
The return type is `Object | false`, not `boolean`, see https://github.com/motdotla/dotenv/blob/v2.0.0/lib/main.js#L36 and https://github.com/motdotla/dotenv/blob/v2.0.0/lib/main.js#L41

### CODE REVIEW

**Comment by vvakame** [comment]
We can't get response from author. please ping me after 3 days. I'll merge this PR.

**Comment by borekb** [comment]
Hopefully I'll remember. Is there a bot for that? :)

**Review by jussikinnula — APPROVED** [review]
Seems to be all in order!

File: dotenv/index.d.ts [hunk 1]
```diff
@@ -1,12 +1,35 @@
 // Type definitions for dotenv 2.0.0
 // Project: https://github.com/motdotla/dotenv
-// Definitions by: Jussi Kinnula <https://github.com/jussikinnula/>
-// Definitions: https://github.com/jussikinnula/DefinitelyTyped
+// Definitions by: Borek Bernard <https://github.com/borekb>
```
Comment (vvakame) [inline dotenv/index.d.ts]: don't remove original author's name. please add your name below.

Comment (borekb) [inline dotenv/index.d.ts]: Good point. Fixed.

**Comment by vvakame** [comment]
LGTM. waiting for Travis CI completed.

**Pull request merged**